lib/vdsm/storage/hsm.py

```diff
@@ -2857,6 +2858,25 @@ def measure(self, sdUUID, imgUUID, volUUID, dest_format, backing=True,
 
         return dict(result=result)
 
+    @public
+    def prune_bitmaps(self, sdUUID, imgUUID, volUUID, baseUUID):
+        """
+        Prune stale bitmaps from the base volume.
+
+        Arguments:
+            sdUUID (str): The UUID of the storage domain that owns the volume.
+            imgUUID (str): The UUID of the image contained on the volume.
+            volUUID (str): The UUID of the top volume.
+            baseUUID (str): The UUID of the volume that will get its bitmaps
+                pruned if missing or invalid in top.
+        """
+        base_vol = self._produce_volume(sdUUID, imgUUID, baseUUID)
+        top_vol = self._produce_volume(sdUUID, imgUUID, volUUID)
```

Comment:
Using _produce_volume() is nice, but it only a shared lock on the storage domain.

We should to take also:
- shared lock on the image? (check how storage jobs lock similar operations)
- shared lock on the top volume
- exclusive lock on the base volume (modified using this code)

Since this should be called only from virt code, and the vm owns the volumes,
no other storage code should access them (engine should ensure this) but it
is safer to take locks so bad engine request will fail at vdsm instead of
corrupting the volume.

You can add locking in a different commit to simplify the work.